aderyn_core/src/detect/experimental/immediate_children.rs

```diff
@@ -0,0 +1,98 @@
+#![allow(clippy::collapsible_match)]
+use std::{collections::BTreeMap, error::Error};
+
+use crate::{
+    ast::{NodeID, NodeType},
+    capture,
+    context::{browser::GetImmediateChildren, workspace_context::WorkspaceContext},
+    detect::detector::{IssueDetector, IssueDetectorNamePool, IssueSeverity},
+};
+use eyre::Result;
+
+#[derive(Default)]
+pub struct ImmediateChildrenDemonstrator {
+    // Keys are source file name and line number
+    found_instances: BTreeMap<(String, usize, String), NodeID>,
+}
+
+impl IssueDetector for ImmediateChildrenDemonstrator {
+    fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
+        // Step 1: Find the ParentChain contract
+
+        let parent_chain_contract = context
+            .contract_definitions()
+            .into_iter()
+            .filter(|x| x.name.contains("ParentChain"))
+            .take(1)
+            .next()
+            .unwrap();
+
+        // Step 2: Find the `increment` function
+
+        let inc = parent_chain_contract
+            .function_definitions()
+            .into_iter()
+            .filter(|x| x.name.contains("increment"))
+            .take(1)
+            .next()
+            .unwrap();
+
+        if let Some(children) = inc.body.as_ref().unwrap().immediate_children(context) {
+            for child in children {
+                assert!(
+                    child.node_type() == NodeType::IfStatement,
+                    "Only if statement should be caught in function body's immediate children!"
```

Comment:
Here, `inc.body` is a function block:
That has a for loop and an if statement nested in it. However, when we call `immediate_children(context)` on the `&Block` node, we only get the for loop